src/dyn/dyn_decoherence_methods.cpp

```diff
@@ -1559,13 +1671,25 @@ void propagate_half_xf(dyn_variables& dyn_var, nHamiltonian& Ham, dyn_control_pa
     CMATRIX Hxf(nadi, nadi);
     CMATRIX D(nadi, nadi); /// this is \exp[-idt/4\hbar * ( T_new.H()*Hxf(t+dt)*T_new + Hxf(t) )]
 
-    XF_correction(Hxf_old, dyn_var, C, prms.wp_width, T, itraj);
-    XF_correction(Hxf, dyn_var, C, prms.wp_width, T, itraj);
+    double wp_width;
+    if (prms.use_td_width == 1){
+      wp_width = compute_wp_width(dyn_var, prms, params, dt);

```

Comment:
also, in this function - using "prms" and "params" - seems redundant

Reply:
consult with the comment above.